lib/kernel/src/prim_tty.erl

```diff
@@ -611,29 +640,77 @@ handle_request(State = #state{unicode = U, cols = W}, redraw_prompt_pre_deleted)
+                            ExpandRows = (cols_multiline(BufferExpandLines, W, U) div W),
+                            ExpandRowsLimit = case State#state.buffer_expand_limit of
+                                0 ->
+                                     ExpandRows;
+                                Limit ->
+                                    min(Limit, ExpandRows)
+                            end,
+                            ExpandRowsLimit1 = min(ExpandRowsLimit, R-1-InputRows),
+                            BufferExpand1 = case ExpandRows > (R-InputRows) of
+                                true ->
+                                        Color = ansi_color(cyan, bright_white),
+                                        StatusLine = io_lib:format(Color ++"\e[1m" ++ "rows ~w to ~w of ~w" ++ "\e[0m",
+                                                                   [ERow, (ERow-1) + ExpandRowsLimit1, ExpandRows]),
+                                        Cols1 = max(0,W*ExpandRowsLimit1),
+                                        Cols0 = max(0,W*(ERow-1)),
+                                        {_, _, BufferExpandLinesInViewStart, {_, BEStartIVHalf}} = split_cols_multiline(Cols0, BufferExpandLines, U, W),
+                                        {_, BufferExpandLinesInViewRev, _, {BEIVHalf, _}} = split_cols_multiline(Cols1, BufferExpandLinesInViewStart++[BEStartIVHalf], U, W),
+                                        BEIVHalf1 = case BEIVHalf of [] -> [];
+                                            _ -> [BEIVHalf]
+                                        end,
+                                        ExpandInView = lists:reverse(BEIVHalf1++BufferExpandLinesInViewRev),
+                                        ["\r\n",lists:join("\n", ExpandInView ++ [StatusLine])];
+                                false ->
+                                    ["\r\n",BufferExpand]
+                                end,
+                          {ExpandBuffer, NewState} = insert_buf(RedrawState#state{ buffer_expand = [] }, iolist_to_binary(BufferExpand1)),
                           BECols = cols(W, End, NewState#state.buffer_expand, U),
                           MoveToEnd = move_cursor(RedrawState, BECols, End),
                           {[encode(Redraw,U),encode(ExpandBuffer, U), MoveToEnd, Movement], RedrawState}
                   end,
     {Output, State};
+handle_request(State = #state{ buffer_expand = Expand, buffer_expand_row = ERow, cols = W, rows = WindowRows, unicode = U}, {move_expand, N}) ->
+    %% Get number of Lines in terminal window
+    BufferExpandLines = case Expand of
+        undefined -> [];
+        _ -> string:split(erlang:binary_to_list(Expand), "\n", all)
+    end,
+    ExpandRows = (cols_multiline(BufferExpandLines, W, U) div W),
+    InputRows = (cols_multiline([State#state.buffer_before ++ State#state.buffer_after], W, U) div W),
+    ERow1 = if ExpandRows > WindowRows-InputRows -> %% We need to page expand rows
```

Comment:
Something is off with this calculation. When the number of rows is less than windows size but larger than 7 the navigation does not work.

To see the effect type `l` `<TAB>`  and then try to navigate.